src/peep/peep.c

```diff
@@ -490,6 +492,7 @@ static void sub_68F41A(rct_peep *peep, int index)
 
 				if (peep->guest_heading_to_ride_id == 0xFF){
 					peep->happiness_growth_rate = max(peep->happiness_growth_rate - 128, 0);
+					if (peep->peep_flags & PEEP_FLAGS_TRACKING) { return; }
```

Comment:
I am not sure you can just return here without consequences... look at the surrounding logic. It should probably instead wrap the next three lines.